tests/components/abode/test_switch.py

```diff
@@ -0,0 +1,125 @@
+"""Tests for the Abode switch device."""
+from unittest.mock import patch
+
+from homeassistant.components.abode import (
+    DOMAIN as ABODE_DOMAIN,
+    SERVICE_TRIGGER_AUTOMATION,
+)
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+
+from .common import setup_platform
+
+AUTOMATION_ID = "switch.test_automation"
+AUTOMATION_UID = "47fae27488f74f55b964a81a066c3a01"
+DEVICE_ID = "switch.test_switch"
+DEVICE_UID = "0012a4d3614cb7e2b8c9abea31d2fb2a"
+
+
+async def test_entity_registry(hass):
+    """Tests that the devices are registered in the entity registry."""
+    await setup_platform(hass, SWITCH_DOMAIN)
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    entry = entity_registry.async_get(AUTOMATION_ID)
+    assert entry.unique_id == AUTOMATION_UID
+
+    entry = entity_registry.async_get(DEVICE_ID)
+    assert entry.unique_id == DEVICE_UID
+
+
+async def test_attributes(hass):
+    """Test the switch attributes are correct."""
+    await setup_platform(hass, SWITCH_DOMAIN)
+
+    state = hass.states.get(DEVICE_ID)
+    assert state.state == STATE_OFF
+
+
+async def test_switch_on(hass):
+    """Test the switch can be turned on."""
+    await setup_platform(hass, SWITCH_DOMAIN)
+
+    with patch("abodepy.AbodeSwitch.switch_on") as mock_switch_on:
+        assert await hass.services.async_call(
+            SWITCH_DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: DEVICE_ID}, blocking=True
+        )
+        await hass.async_block_till_done()
+
+        mock_switch_on.assert_called_once()
+
+
+async def test_switch_off(hass):
+    """Test the switch can be turned off."""
+    await setup_platform(hass, SWITCH_DOMAIN)
+
+    with patch("abodepy.AbodeSwitch.switch_off") as mock_switch_off:
+        assert await hass.services.async_call(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5184102)